pkg/server/server.go

```diff
@@ -105,45 +99,9 @@ func (s *server) RunUntil(stopCh <-chan struct{}) error {
 		return nil
 	}
 
-	// Start serving API and scrape loop
-	go s.runScrape(ctx)
 	return s.GenericAPIServer.PrepareRun().Run(stopCh)
 }
 
-func (s *server) runScrape(ctx context.Context) {
-	ticker := time.NewTicker(s.resolution)
-	defer ticker.Stop()
-	s.tick(ctx, time.Now())
-
-	for {
-		select {
-		case startTime := <-ticker.C:
-			s.tick(ctx, startTime)
-		case <-ctx.Done():
-			return
-		}
-	}
-}
-
-func (s *server) tick(ctx context.Context, startTime time.Time) {
-	s.tickStatusMux.Lock()
-	s.tickLastStart = startTime
```

Comment:
Correct me if I am wrong.
In this implementation, `probeMetricCollectionTimely()` will always return an error, right?